### PR TITLE
Clone repository with gittuf namespaces and push RSL and policy namespaces to remote

### PR DESCRIPTION
**Pull request by adityasaky**
Based on https://github.com/gittuf/gittuf/pull/115#issuecomment-1756038262.

This PR adds a clone command that clones a repository from the specified remote location with its gittuf namespaces. Additionally, it implements a push command for the policy and RSL namespaces. The remote subcommand for `gittuf policy` and `gittuf trust` are shared in a new package.

### CODE REVIEW

File: internal/repository/sync.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	ErrDirExists         = errors.New("directory exists")
+)
+
+// Clone wraps a typical git clone invocation, fetching gittuf refs in addition
+// to the standard refs. It performs a verification of the RSL against the
+// specified HEAD after cloning the repository.
+// TODO: resolve how root keys are trusted / bootstrapped.
+func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Repository, error) {
+	if dir == "" {
+		// FIXME: my understanding is backslashes are not used in URLs but I haven't dived into the RFCs to check yet
```
Comment (adityasaky) [inline internal/repository/sync.go:26]: This comment is still outstanding @wlynch.

**Comment by adityasaky** [comment]
Okay, I think it's better to do this in much smaller chunks than the last couple times :smile: 

This PR has clone and push functions. I think fetch can happen separately because we had some gitinterface changes in #115 that I'll submit separately to make this easier. In the meantime, this should be good for review!

**Review by wlynch — APPROVED** [review]

File: internal/repository/sync.go (hunk 1 above)
Comment (adityasaky) [reply internal/repository/sync.go:28]: Opened an issue to track this. https://github.com/gittuf/gittuf/issues/140

File: internal/cmd/clone/clone.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		"Specify branch to check out",
+	)
+}
+
+func (o *options) Run(cmd *cobra.Command, args []string) error {
+	var dir string
+	if len(args) > 1 {
+		dir = args[1]
+	}
+	_, err := repository.Clone(context.Background(), args[0], dir, o.branch)
```
[the remark this answers is not in the document]
Comment (adityasaky) [reply internal/cmd/clone/clone.go]: Done!

**Comment by adityasaky** [comment]
Oh I missed this was already approved barring the change on ctx :sweat_smile:

**Pull request merged**